calour/io.py

```diff
@@ -432,131 +519,95 @@ def read_open_ms(data_file, sample_metadata_file=None, gnps_file=None, feature_m
 
     Returns
     -------
-    Experiment
+    MS1Experiment
 
     See Also
     --------
     read
     '''
-    logger.debug('Reading OpenMS data (OpenMS bucket table %s, map file %s)' % (data_file, sample_metadata_file))
-    if rows_are_samples:
-        data_file_type = 'openms_transpose'
+    if data_table_params is None:
+        data_table_params = {}
+    if data_file_type == 'mzmine2':
+        param_defaults = {'sample_in_row': False, 'data_table_sep': ',', 'direct_ids': True, 'cut_sample_id_sep': '_'}
+        ctype = 'csv'
+    elif data_file_type == 'biom':
+        param_defaults = {'sample_in_row': False, 'direct_ids': False, 'get_mz_rt_from_feature_id': True}
+        ctype = 'biom'
+    elif data_file_type == 'openms':
+        param_defaults = {'sample_in_row': False, 'data_table_sep': ',', 'direct_ids': False, 'get_mz_rt_from_feature_id': True}
+        ctype = 'csv'
+    elif data_file_type == 'gnps-ms2':
+        param_defaults = {'sample_in_row': False, 'direct_ids': True}
+        ctype = 'biom'
     else:
-        data_file_type = 'openms'
+        raise ValueError("ms data_file_type %s not supported. please use ['mzmine2','openms','gnps-ms2']")
+    param_defaults['use_gnps_id_from_AllFiles'] = True
+    param_defaults['mz_rt_sep'] = '_'
+    # set the default param values if not supplied by the user
+    for cparam, cval in param_defaults.items():
+        data_table_params[cparam] = data_table_params.get(cparam, cval)
+
```

Comment:
this is even more verbose than before. plus, the func param `data_table_params` is ugly.

Reply:
fixed.
if you have a better less ugly name - i can update.